staging/src/k8s.io/code-generator/cmd/register-gen/generators/register_external.go

```diff
@@ -61,11 +61,13 @@ func (g *registerExternalGenerator) Finalize(context *generator.Context, w io.Wr
 
 	sw := generator.NewSnippetWriter(w, context, "$", "$")
 	m := map[string]interface{}{
-		"groupName":         g.gv.Group,
-		"version":           g.gv.Version,
-		"types":             typesToGenerateOnlyNames,
-		"addToGroupVersion": context.Universe.Function(types.Name{Package: "k8s.io/apimachinery/pkg/apis/meta/v1", Name: "AddToGroupVersion"}),
-		"groupVersion":      context.Universe.Type(types.Name{Package: "k8s.io/apimachinery/pkg/apis/meta/v1", Name: "GroupVersion"}),
+		"groupName":          g.gv.Group,
+		"version":            g.gv.Version,
+		"types":              typesToGenerateOnlyNames,
+		"addToGroupVersion":  context.Universe.Function(types.Name{Package: "k8s.io/apimachinery/pkg/apis/meta/v1", Name: "AddToGroupVersion"}),
+		"groupVersion":       context.Universe.Type(types.Name{Package: "k8s.io/apimachinery/pkg/apis/meta/v1", Name: "GroupVersion"}),
+		"runtimeScheme":      context.Universe.Type(types.Name{Package: "k8s.io/apimachinery/pkg/runtime", Name: "Scheme"}),
```

Comment:
> @sttts when I run register-gen from this script the import tracker doesn't import k8s.io/apimachinery/pkg/runtime and k8s.io/apimachinery/pkg/runtime/schema but they are imported when I run it manually. Do you know how to fix it ?

adding `runtime` and `schema` explicitly to the import tracker helped to resolve my issue.

It worked when run manually because `golang.org/x/tools/imports` we use to add imports was able to properly find the packages. Unfortunately when `register-gen` is run from the script, the current working directory is different and  `tools/imports` is unable to find the packages.